api/src/main/java/org/openmrs/module/patientflags/db/hibernate/HibernateFlagDAO.java

```diff
@@ -69,7 +69,23 @@ public void setSessionFactory(DbSessionFactory sessionFactory) {
 	public Flag getFlag(Integer flagId) {
 		return (Flag) sessionFactory.getCurrentSession().get(Flag.class, flagId);
 	}
-	
+
+	public Flag getFlagByUuid(String uuid) throws DAOException {
+		return (Flag)this.sessionFactory.getCurrentSession().createQuery("from Flag f where f.uuid = :uuid").setString("uuid", uuid).uniqueResult();
+	}
+
+	public Flag getFlagByName(String name) throws DAOException {
+		Criteria criteria = sessionFactory.getCurrentSession().createCriteria(Flag.class);
+		criteria.add(Restrictions.ilike("name", name, MatchMode.EXACT));

```

Comment:
ilike is never using a db index thus it will be very slow if you happen to have hundreds of calls to this method and/or hundreds of flags... Note that MySQL is creating tables with case insensitive collation, thus you can use eq for case insensitive selects. In core we use the following: https://github.com/openmrs/openmrs-core/blob/571f3080b3d870dd7c1d3a00a01aab1fd3fe2295/api/src/main/java/org/openmrs/api/db/hibernate/HibernateConceptDAO.java#L1043-L1047 If you happen to use a case sensitive collation in MySQL (not default) then you set the GP.